File: 2259-remove-digit-from-number-to-maximize-result/2259-remove-digit-from-number-to-maximize-result.cpp
class Solution {
public:
    string removeDigit(string number, char digit) {
        string maxm = "";
        for(int i=0; i<number.length(); i++){
            if(number[i]==digit){
                string temp = number.substr(0,i)+number.substr(i+1, number.length()-1); 
                maxm = max(maxm, temp);
            }
        }
        return maxm;
    }
};

File: 2259-remove-digit-from-number-to-maximize-result/2259-remove-digit-from-number-to-maximize-result.java
class Solution {
    public String removeDigit(String number, char digit) {
        char[] cs = number.toCharArray();
        String ans = "";
        for(int i=0; i<cs.length-1; i++){
            if(cs[i]==digit){
                if(cs[i]<cs[i+1])
                    return String.valueOf(number.substring(0, i)+number.substring(i+1));
            }
        }
        int li = number.lastIndexOf(digit);
        return String.valueOf(number.substring(0, li)+number.substring(li+1));
    }
}